### ISSUE TITLE
Add warning to docs that build directory is removed. 

### ISSUE THREAD

**Issue by kescobo** [JuliaDocs/Documenter.jl#1012]
I was recently playing around with the `macdocs()` options, and set the root directory as the build directory. On running the function, my directory was wiped out, including the `.git` directory, so I lost a bunch of work I hadn't pushed. 

This is of course my bad, but Morten suggested on slack that adding a warning to the function docstring might be a good idea. If someone can think of a way to prevent someone from making my dumb mistake, that would be cool too, but I can't think of an intervention that wouldn't be overly onerous if someone is doing something non-standard on purpose.

**Comment by mortenpi** [comment]
Two thoughts:

* We could have the default `clean = true` be an "clever" version that refuses to remove it some situations and then separately support `clean = :force`, which just indiscriminately deletes everything.
* Not sure what the heuristics should be though. But the standard setup is that the build directory is next to `make.jl`.. so we could maybe check for that?

**Comment by kescobo** [comment]
That makes sense to me. A slightly more complicated heuristic idea I had: if the build directory has file types that documenter doesn't create (eg anything not `html`, `css`, `.js`, and whatever other file types are expected). This would require walking the build directory until it reaches the end or encounters something that doesn't match, but I don't think that would slow things down too much, right?

**Comment by fredrikekre** [comment]
`clean = :force` seems not needed since you can then just `rm -rf` yourself. Maybe `clean = true` should only remove stuff it the directory looks like a documenter output build directory or something?

**Comment by kescobo** [comment]
>clean = :force seems not needed since you can then just rm -rf yourself.

True, but if you know the heuristic doesn't work for you, adding this so that building is 1 step instead of 2 steps still seems like a nice convenience, no?
